[qr0RWWHT] - Цели

KossaDenis opened the pull request:
- Добавил секцию с целями

Commit:
price-card component

Commit:
[O5fNSvsN] Цены
-- добавлен price-card компонент

Commit:
[O5fNSvsN] Цены
- добавил компонент карточки для прайс листа
- реализовал фичу переключения списка прайс листа через таб

Commit:
[O5fNSvsN] Цены
- добавил компонент карточки для прайс листа
- реализовал фичу переключения списка прайс листа через таб

Commit:
Merge branch 'feature/O5fNSvsN' of https://github.com/weekkkk/x-install--landing-page into feature/O5fNSvsN

Commit:
[O5fNSvsN] Цены
-- добавлены градиентные фигуры по бокам блока "Цены"

KossaDenis committed:
[qr0RWWHT] - Цели

KossaDenis committed:
[qr0RWWHT] - Цели

weekkkk commented on src/pages/main/ui/page/sections/purposes/index.vue:
```diff
@@ -0,0 +1,56 @@
+import type { KeyFigures } from "./types";
+
+/**Ключевые показатели */
+const keyFiguresList: KeyFigures[] = [
+  {
+    Id: Date.now(),
+    Title: "20K+",
+    Paragraph: "Загрузок В День",
+  },
+  {
+    Id: Date.now(),
+    Title: "170+",
+    Paragraph: "Стран",
+  },
+  {
+    Id: Date.now(),
+    Title: "2,5M+",
+    Paragraph: "Активных Пользователей",
+  },
+  {
+    Id: Date.now(),
+    Title: "700+",
+    Paragraph: "Клиентов",
+  },
```
аналогично https://github.com/weekkkk/x-install--landing-page/pull/5#discussion_r1984055145

weekkkk commented on src/pages/main/ui/page/sections/purposes/index.vue:
```diff
@@ -0,0 +1,56 @@
+/**Ключевые показатели */
+const keyFiguresList: KeyFigures[] = [
+  {
+    Id: Date.now(),
```
плохой тон задавать id в timestamp, тем более, что тут в этом никакой подтребности, сделай фиксированное значение

weekkkk commented on src/pages/main/ui/page/sections/purposes/index.vue:
```diff
@@ -0,0 +1,56 @@
+      <span
+        class="bg-gradient-to-r dark:from-turquoise-100 dark:to-green bg-clip-text text-transparent"
+      >
+        целевого действия</span
+      >: покупке, регистрации, переходу на сайт
+    </div>
```
поправь форматирование

weekkkk commented on src/pages/main/ui/page/sections/purposes/index.vue:
```diff
@@ -0,0 +1,56 @@
+      >: покупке, регистрации, переходу на сайт
+    </div>
+    <div
+      class="mt-[5.5rem] flex gap-[11.4rem] justify-between ml-[4.7rem] mr-[5.7rem]"
```
проверь еще раз, по идее для `mt-[5.5rem]` должен быть дефолтный класс

KossaDenis replied:
нету

weekkkk commented on src/pages/main/ui/page/sections/purposes/types/type.ts:
```diff
@@ -0,0 +1,5 @@
+export type KeyFigures = {
+  Id: number;
+  Title: string;
+  Paragraph: string;
```
зачем называть с большой буквы, ты не в тирскрипте, большой разницы конечно нет, просто там в коммерции не делают обычно, тем более в проекте все camelCase написано, а название папок и файлов kebab-case

weekkkk replied:
где-то у леши увидел что ты папку назвал PascalCase, исправишь на kebab-case

weekkkk commented on src/pages/main/ui/page/sections/purposes/types/type.ts:
этот тип нужно хранить в компоненте, для которого он используется

KossaDenis replied:
то есть не надо создавать index.ts с экспортом ? а сразу ипортить в компонент где он нужен ?

weekkkk replied:
> то есть не надо создавать index.ts с экспортом ? а сразу ипортить в компонент где он нужен ?

нет, этот тип относится к компоненту KeyFigures, значит хранить его нужно там же

weekkkk commented on src/widgets/KeyFigures/index.vue:
```diff
@@ -0,0 +1,18 @@
+<template>
+  <div class="text-center">
+    <h1
+      class="text-5xl leading-[100%] font-bold bg-gradient-to-r dark:from-turquoise-100 dark:to-green bg-clip-text text-transparent"
```
зачем `leading-[100%]`, когда есть `leading-none`

weekkkk commented on src/widgets/KeyFigures/types/props.ts:
```diff
@@ -0,0 +1,4 @@
+export type KeyFiguresProps = {
+  Title: string;
+  Paragraph: string;
+};
```
camelCase

weekkkk commented on src/widgets/KeyFigures/index.vue:
это не виджет
Виджеты — большие автономные блоки функциональности или пользовательского интерфейса, обычно реализующие целый сценарий использования.
https://feature-sliced.design/docs/get-started/overview
смотри в документацию, если не знаешь что куда пихать

weekkkk replied:
название папки PascalCase не используем, переделай все такое на kebab-case

Commit:
[O5fNSvsN] Цены
-- добавлен экспорт UiPriceCardProps в shared/ui
-- все статичные виджета PriceList данные были пересены в price/ui/list/consts/..
-- добавлен ключ для перебора UiPriceCard

KossaDenis committed:
[qr0RWWHT] - Цель
- Создать секцию целей

KossaDenis committed:
[qr0RWWHT] - Цель
- Создать секцию целей

weekkkk committed:
Merge pull request #5 from weekkkk/feature/O5fNSvsN

[O5fNSvsN] Цены

KossaDenis committed:
[qr0RWWHT] - Цель
- Создать секцию целей

KossaDenis committed:
[qr0RWWHT] - Цель
- Создать секцию целей

weekkkk commented on src/pages/main/ui/page/index.vue:
```diff
@@ -1,6 +1,6 @@
 <template>
-  <LayoutHeaderWidget />
-  <MainPageSectionsWelcome class="mt-[10.65rem] h-[52.55rem]" />
+  <!-- <LayoutHeaderWidget /> -->
+  <!-- <MainPageSectionsWelcome class="mt-[10.65rem] h-[52.55rem]" /> -->
```
раскоменти

weekkkk commented on src/pages/main/ui/page/sections/purposes/index.vue:
```diff
@@ -1,56 +1,61 @@
 <script setup lang="ts">
-import type { KeyFigures } from "./types";
+import type { keyFigureList } from "~/src/shared/ui/key-figures/types";
```
эту константу нужно хранить в папке этой секции

KossaDenis replied:
не понял что имеется ввиду

weekkkk replied:
список `keyFigureList ` нужно хранить в `sections/purposes/consts`

weekkkk commented on src/shared/ui/key-figures/types/props.ts:
```diff
@@ -1,4 +1,4 @@
+export type keyFiguresProps = {
   Title: string;
   Paragraph: string;
 };
```
уже писал, нейминг поправь, PaskalCase использует только для названий моделей, свойства называем camelCase, файлы называем kebab-case

KossaDenis committed:
[qr0RWWHT] - Цели
- Исправления + адаптив

KossaDenis committed:
[qr0RWWHT] - Цели
- Исправления + адаптив

Commit:
[O5fNSvsN] Цены
-- адаптив компонента и виджета

Commit:
[O5fNSvsN] Цены
-- правки по функционалу

KossaDenis committed:
[qr0RWWHT] - Цели
- Правки

KossaDenis committed:
[qr0RWWHT] - Цели
- Правки

Commit:
[O5fNSvsN] Цены
-- адаптив для компонента PriceCard
-- правки по фунционалу в логике отрисовки под мобилки,
добавлен один необязательный props для PriceCard
md: {guarantees?: string[]}

KossaDenis committed:
[qr0RWWHT] - Цель

KossaDenis committed:
[qr0RWWHT] - Цель

KossaDenis committed:
[qr0RWWHT] - Цель
- Правки

KossaDenis committed:
[qr0RWWHT] - Цель
- Правки

weekkkk commented on src/pages/main/ui/page/sections/purposes/consts/const.ts:
почему тип лежит в папке констант

weekkkk commented on src/pages/main/ui/page/sections/purposes/index.vue:
```diff
@@ -0,0 +1,61 @@
+import type { KeyFigureList } from "./consts";
+
+/**Ключевые показатели */
+const keyFiguresList: KeyFigureList[] = [
+  {
+    id: 1,
+    title: "20K+",
+    paragraph: "Загрузок В День",
+  },
+  {
+    id: 2,
+    title: "170+",
+    paragraph: "Стран",
+  },
+  {
+    id: 3,
+    title: "2,5M+",
+    paragraph: "Активных Пользователей",
+  },
+  {
+    id: 4,
+    title: "700+",
+    paragraph: "Клиентов",
+  },
+];
```
этот список нужно вынести в констану в отдельном файле, в папке const

weekkkk committed:
Merge pull request #10 from weekkkk/feature/O5fNSvsN

[O5fNSvsN] Цены

KossaDenis reviewed (commented):
- правки

KossaDenis committed:
[qr0RWWHT] - Цели
- Правки
